File: part1-coding/07-heap-meeting-rooms/python/07_heap_meeting_rooms.py
import heapq
def min_meeting_rooms(intervals):
    intervals.sort(key=lambda x:x[0])
    h=[]
    for s,e in intervals:
        if h and h[0] <= s: heapq.heappop(h)
        heapq.heappush(h,e)
    return len(h)

File: part1-coding/07-heap-meeting-rooms/cpp/07-heap-meeting-rooms.cpp
#include <bits/stdc++.h>
using namespace std;
int minMeetingRooms(vector<vector<int>>& in){
    sort(in.begin(), in.end(), [](auto&a,auto&b){return a[0]<b[0];});
    priority_queue<int, vector<int>, greater<int>> pq;
    for(auto &m: in){
        if(!pq.empty() && pq.top() <= m[0]) pq.pop();
        pq.push(m[1]);
    }
    return (int)pq.size();
}
int main(){ return 0; }
